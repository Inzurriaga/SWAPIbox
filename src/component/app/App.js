import React, { Component } from 'react';
import Intro from "../intro/Intro"
import MainPage from "../mainpage/MainPage"


class App extends Component {
  constructor(){
    super();
    this.state = {
      intro: true
    }
  }
  render() {
    const { intro } = this.state
    return (
      <div className="App">
       {
         intro ? 
          <Intro />:
          <MainPage />
       }
      </div>
    );
  }
}

export default App;
