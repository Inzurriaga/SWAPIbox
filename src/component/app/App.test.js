import React from 'react';
import { shallow } from "enzyme"
import App from './App';

describe("app", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  })
  it("should have a snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("should have a default state", () => {
    expect(wrapper.state("intro")).toEqual(true)
  })
  it("should change the state of intro to false", () => {
    expect(wrapper.state("intro")).toEqual(true)
    wrapper.instance().exitIntro()
    expect(wrapper.state("intro")).toEqual(false)
  })
})