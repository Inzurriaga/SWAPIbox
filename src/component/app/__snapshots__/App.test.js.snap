// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should have a snapshot 1`] = `
<div
  className="App"
>
  <Intro
    exitIntro={[Function]}
  />
</div>
`;
